restli-server/src/main/java/com/linkedin/restli/internal/server/util/MIMEParse.java

```diff
@@ -67,12 +67,25 @@
     @Override
     public String toString()
     {
-      StringBuffer s = new StringBuffer("('" + type + "', '" + subType
-                                                + "', {");
+      StringBuilder s = new StringBuilder("('").append(type).append("', '").append(subType).append("', {");
       for (String k : params.keySet())
-        s.append("'" + k + "':'" + params.get(k) + "',");
+        s.append("'").append(k).append("':'").append(params.get(k)).append("',");
       return s.append("})").toString();
     }
+
+    /**
+     * Build the String for the content type. "q" param is ignored
+     */
+    String toContentType()
```

Comment:
This has a lot of duplication with the toString() method. Why not parameterize this method adding a boolean param like excludeQuality and invoke accordingly?

Reply:
`toString` is a representation of the ParseResult, while `toContentType` is the actual mime type, so I think it's better to keep them separate. The string format is a little different, so I don't see much that can be generalized into a helper.